wolff/optimization/gradient_descent.py

```diff
@@ -0,0 +1,46 @@
+'''Gradient Descent Step'''
+from wolff.layer.layer import Layer
+
+
+class GradientDescent(Layer):
+
+    def __init__(self, theta=[], grad_J=[], learning_rate=0.001):
+        super(GradientDescent, self).__init__()
+
+        self.theta = theta
+        self.grad_J = grad_J
+        self.learning_rate = learning_rate
+
+    @property
+    def theta(self):
+        return self._theta
+
+    @theta.setter
+    def theta(self, value):
+        self._theta = value
+
+    @property
+    def learning_rate(self):
+        return self._learning_rate
+
+    @learning_rate.setter
+    def learning_rate(self, value):
+        if value <= 0:
+            raise ValueError('Constant must be greater than zero')
+        self._learning_rate = value
+
+    @property
+    def grad_J(self):
+        return self._grad_J
+
+    @grad_J.setter
+    def grad_J(self, value):
+        self._grad_J = value
```

Comment:
Since this is a list, we need a deep copy.
```suggestion
        self._grad_J = list(value)
```